client_script/profile_reader.pl

```diff
@@ -198,6 +200,98 @@ sub printkv {
   }
 }
 
+sub checkXsand {
+  my $xsandContent = `pgrep xsand`;
+  if ($xsandContent eq '') {
+    return 'Not running';
+  } else {
+    return 'Running';
+  }
+}
+
+sub checkMounts {
+  my $mountsContent = `mount | grep acfs`;
+  my $mountInt=1;
+  my $data = {};
+  foreach(split(/\n/, $mountsContent)){
+	my @fields = split / /, $_;
+	my @fieldsParts = split /\//, $fields[2];
+    $data->{"$mountInt"}->{"mountPath"}="$fields[2]";
+    $data->{"$mountInt"}->{"name"}="$fieldsParts[-1]";
+    $mountInt = $mountInt + 1;
+  }
+  my $arrayOutput=[];
+  foreach(keys %$data){
+    my $updatedHash = $data->{$_};
+    $updatedHash->{"number"} = $_;
+    push @$arrayOutput, $updatedHash;
+  }
+  return $arrayOutput;
+}
+
+sub pingMetaDataControllers {
+  my $xsanNameServersConfigFile = read_file('/Library/Preferences/Xsan/fsnameservers');
+  my $data = {};
+  my $pingInt=1;
+  my $percentagePacketLoss = 0;
+  foreach(split(/\n/, $xsanNameServersConfigFile)){
+    my $pingContent = `ping -c 1 $_`;
+    my @pingContentArray = split /\n/, $pingContent;
+    my $pingStatus;
+    my $packetLoss;
+    my $pingResult;
+	no warnings 'uninitialized';
+	no warnings 'substr';
+	try {
+	  $pingResult = substr $pingContentArray[4], 23, 1;
```

Comment:
ouch, not sure i like this; any adjustment in the ping output would break it. Could we use a regex instead?